src/main/java/uk/ac/cam/cl/dtg/segue/api/QuestionFacade.java

```diff
@@ -232,6 +261,69 @@ public Response getQuestionsAnswered(@Context final HttpServletRequest request,
     }
   }
 
+  /**
+   * REST end point to provide five questions in random located in the question tile of the student dashboard.
+   *
+   * @param request  - this allows us to check to see if a user is currently logged in.
+   * @param subjects - a comma separated list of subjects
+   * @return a Response containing a gameboard object or containing a SegueErrorResponse.
+   */
+  @GET
+  @Path("/randomQuestions")
+  @Produces(MediaType.APPLICATION_JSON)
+  @GZIP
+  public final Response getRandomQuestions(@Context final HttpServletRequest request,
+                                           @QueryParam("subjects") final String subjects) {
+    try {
+      RegisteredUserDTO currentUser = this.userManager.getCurrentRegisteredUser(request);
+      var filterQuestionsPreference = this.userPreferenceManager.getUserPreference(
+          "DISPLAY_SETTING",
+          "HIDE_NON_AUDIENCE_CONTENT",
+          currentUser.getId());
+
+      GameFilter gameFilter = new GameFilter();
+
+      if (filterQuestionsPreference != null && filterQuestionsPreference.getPreferenceValue()) {
+        var userContexts = currentUser.getRegisteredContexts();
+
+        List<String> subjectsList = splitCsvStringQueryParam(subjects);
+        List<String> stagesList = new ArrayList<>();
+        List<String> examBoardsList = new ArrayList<>();
+
+        for (UserContext uc : userContexts) {
+          stagesList.add(uc.getStage().name());
+          examBoardsList.add(uc.getExamBoard().name());
+        }
+
+        gameFilter = new GameFilter(
+            subjectsList,
+            null,
+            null,
+            null,
+            null,
+            null,
+            stagesList,
+            null,
+            examBoardsList);
+      }
+
+      var questions = this.gameManager.generateRandomQuestions(gameFilter, 5);
+
+      // Return the list of random questions as JSON
+      return Response.ok(questions).build();
+    } catch (NoUserLoggedInException e) {
+      return SegueErrorResponse.getNotLoggedInResponse();
+    } catch (ContentManagerException e) {
+      return new SegueErrorResponse(Status.NOT_FOUND, "Error creating random questions")
+          .toResponse();
+    } catch (SegueDatabaseException e) {
+      SegueErrorResponse error = new SegueErrorResponse(Status.INTERNAL_SERVER_ERROR,
+          "Error while getting user preferences", e);
```

Comment:
In a previous standup, it was mentioned that if the user preferences were set, we can still return some questions. I.e, this shouldn't be a 500 error. Doesn't that mean that this needs changing?

Reply:
Yes the ideal behaviour should be to check for user contexts / user preferences and filter accordingly, but in the event there are no contexts, or no preferences, then it should just be a random 5 questions returned. I think the only time we want an error (and not status 500!) returned to the front end is if there is literally not 5 questions to return for some reason.

Reply:
If there are no questions generated, we throw a 404. 

As Hannah said, if the preference isn't set, or we get a null response, we still generate 5 random questions, or throw a 404 if it's unable to generate 5 questions (impossible without truncating the question tables in the db).

The SegueDatabaseException only happens when there's an issue getting the user preferences. Given what I said earlier, this will only be triggered if we have some kind of Network/infra issue with the DB (network down or DB is literally on fire).

Reply:
Thank you, yes, in that case it's OK to display a 500 error, as it's a legitimate server error from which we cannot recover.
If there are less than 5 questions being generated (for valid reasons), I think it would be better to simply return as many as possible and let the frontend decide (see my other comment).